python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1338,19 +1237,15 @@ def _get_failed_and_downstream_asset_graph_subset(
         asset_graph_view,
         lambda candidate_asset_graph_subset, _: (
             AssetGraphViewBfsFilterConditionResult(
-                passed_asset_graph_subset=_get_subset_in_target_subset(
-                    asset_graph_view,
-                    candidate_asset_graph_subset,
-                    asset_backfill_data.target_subset,
-                ),
+                passed_asset_graph_subset=candidate_asset_graph_subset,
                 excluded_asset_graph_subsets_and_reasons=[],
             )
         ),
         initial_asset_graph_subset=failed_asset_graph_subset,
         include_full_execution_set=False,
     )[0]
 
-    return failed_and_downstream_subset
+    return failed_and_downstream_subset & asset_backfill_data.target_subset
```

Comment:
hm so lets say you backfill a single asset in the target subset at the top of the graph and it fails, and it comes time to do this calculation.

Before, it would be done pretty quickly, because it is intersecting with the target subset at each bfs step.

Now, it would traverse the entire graph, then do an intersection at the end. That seems potentially significant even with subsets in the mix?

Or is the idea that later in the stack we are going to somehow do away with bfs entirely so this is a moot point?